Do not wait for events to be dispatched to IDBRequest

szewai opened the pull request:
The test expects event to be dispatched to IDBRequest after its execution context is stopped, but this is not standard behavior. WebKit and Blink can stop dispatching event as soon as iframe is detached (iframe.remove() is invoked),
which leads to test timeout. Since the goal of the test is to verify the accessor is still valid after execution context is destroyed (no exception is thrown), we can just check readyState without wait.

szewai commented:
The problem is mentioned in https://github.com/w3c/IndexedDB/issues/416.
I verified the updated test passes in Chrome, Firefox and Safari.

inexorabletash commented on IndexedDB/ready-state-destroyed-execution-context.html:
```diff
@@ -21,11 +22,8 @@
-      openRequest.onsuccess = resolve;
-    });
-    assert_equals(openRequest.readyState, 'done');
+    await load_iframe();
```
Do we need to await _anything_ here?

The original test was authored here:

https://chromium-review.googlesource.com/c/chromium/src/+/3125334

... and the intention was just to ensure that readyState() could be called without hitting a CHECK in Chromium after the context was destroyed. 

In the PR description, you say _"make the test wait on a new iframe with the same id to be loaded, so that there is enough time for request to be completed."_ - but we don't actually care about the request being completed -- or we'd assert that it was 'done'.

So what about just removing the await entirely (added in https://phabricator.services.mozilla.com/D203270) and just keeping the assert that `readyState` returns a string (i.e. keep the next line as you suggest)

szewai replied:
Yes, I think the wait is unnecessary if we don't care about the value. I will update the PR.

inexorabletash reviewed (commented):
Thanks for this!

szewai committed:
Do not wait for events to be dispatched to IDBRequest

The test expects event to be dispatched to IDBRequest after its execution
context is stopped, but this is not standard behavior. WebKit and Blink can stop
dispatching event as soon as iframe is detached (iframe.remove() is invoked),
which leads to test timeout. Since the goal of the test is to verify the accessor
is still valid after execution context is destroyed (no exception is thrown), we
can just check readyState without wait. We only check the type instead of value
as different engines handle the destruction differently, so the state may not be
updated right away.

inexorabletash reviewed (approved):
LGTM!
